server/src/test/java/org/elasticsearch/tasks/ListTasksResponseTests.java

```diff
@@ -23,78 +23,138 @@
-        return false;
+        return true;
+    }
+
+    @Override
+    protected Predicate<String> getRandomFieldsExcludeFilter() {
+        //status and headers hold arbitrary content, we can't inject random fields in them
+        return field -> field.endsWith("status") || field.endsWith("headers");
     }
 
     @Override
     protected void assertEqualInstances(ListTasksResponse expectedInstance, ListTasksResponse newInstance) {
         assertNotSame(expectedInstance, newInstance);
         assertThat(newInstance.getTasks(), equalTo(expectedInstance.getTasks()));
-        assertThat(newInstance.getNodeFailures().size(), equalTo(1));
-        for (ElasticsearchException failure : newInstance.getNodeFailures()) {
-            assertThat(failure, notNullValue());
-            assertThat(failure.getMessage(), equalTo("Elasticsearch exception [type=failed_node_exception, reason=message]"));
+        assertThat(newInstance.getNodeFailures().size(), equalTo(expectedInstance.getNodeFailures().size()));
+        for (int i = 0; i < newInstance.getNodeFailures().size(); i++) {
+            ElasticsearchException newException = newInstance.getNodeFailures().get(i);
+            ElasticsearchException expectedException = expectedInstance.getNodeFailures().get(i);
+            assertThat(newException.getMetadata("es.node_id").get(0), equalTo(((FailedNodeException)expectedException).nodeId()));
+            assertThat(newException.getMessage(), equalTo("Elasticsearch exception [type=failed_node_exception, reason=error message]"));
+            assertThat(newException.getCause(), instanceOf(ElasticsearchException.class));
+            ElasticsearchException cause = (ElasticsearchException) newException.getCause();
+            assertThat(cause.getMessage(), equalTo("Elasticsearch exception [type=connect_exception, reason=null]"));
+        }
+        assertThat(newInstance.getTaskFailures().size(), equalTo(expectedInstance.getTaskFailures().size()));
+        for (int i = 0; i < newInstance.getTaskFailures().size(); i++) {
+            TaskOperationFailure newFailure = newInstance.getTaskFailures().get(i);
+            TaskOperationFailure expectedFailure = expectedInstance.getTaskFailures().get(i);
+            assertThat(newFailure.getNodeId(), equalTo(expectedFailure.getNodeId()));
+            assertThat(newFailure.getTaskId(), equalTo(expectedFailure.getTaskId()));
+            assertThat(newFailure.getStatus(), equalTo(expectedFailure.getStatus()));
+            assertThat(newFailure.getCause(), instanceOf(ElasticsearchException.class));
+            ElasticsearchException cause = (ElasticsearchException) newFailure.getCause();
+            assertThat(cause.getMessage(), equalTo("Elasticsearch exception [type=illegal_state_exception, reason=null]"));
         }
     }
 
-    @Override
-    protected boolean assertToXContentEquivalence() {
-        return false;
+    public void testFromXContentWithFailures() throws IOException {
```

Comment:
nit: can you add a comment what the actual goal of this test is, and what the failure scenario is?
I kind of figured it out from the comment inside I think, but having it as a method comment would speed up reading on later occasions. 
From what I got so far: we don't really want to test that parsing works (i.e. we don't assert anything on the result), but only that it doesn' break. The failure source in this case would be random fields inside the task/nodefailures objects, thats why those are not mentioned in `getRandomFieldsExcludeFilter`
Anything I'm missing? Is this more or less correct?

Reply:
I will add a comment, we actually assert on everything, this is a normal testFromXContent with the only difference that we disabled assertToXContentEquivalence ;)

Reply:
I thought I knew how to read these tests but apparently I missed that. See what you mean now. Even one more reason to make life easier for future-me by adding a comment. Thanks.